### DISCUSSION TITLE
I am experiencing an issue with Wezterm where it creates files like `gui-sock-12888` in the `~/.local/share/wezterm` directory even though my config file `wezterm.lua` is located in `~/.config/wezterm`.

### DISCUSSION THREAD

**Discussion by khanra17** [wezterm/wezterm#4422] [Q&A]
### What Operating System(s) are you running on?

Windows

### WezTerm version

20230712-072601-f4abf8fd

### Ask your question!

I have set up my Wezterm configuration in accordance with the XDG Base Directory Specification by placing my wezterm.lua file in the ~/.config/wezterm directory and it works great. However, when I open the terminal, Wezterm creates files like gui-sock-12888 in the ~/.local/share/wezterm directory.

I have also attempted to set the XDG_RUNTIME_DIR environment variable, but this did not resolve the issue.

If there is a way to configure Wezterm to use a different directory for these files via environment variables or any other workaround, please let me know.

Thank you for your assistance in resolving this issue.

**Comment by tpoliaw** [comment]
I don't have access to Windows so can't test it but `~/.local/share` is the default for `$XDG_DATA_HOME` so setting that might work.

Are you trying to get it to create the sock file in the config directory as well? I'm not sure that really make sense given it's not configuration. ~~The XDG spec says "XDG_DATA_HOME defines the base directory relative to which user specific data files should be stored" so it makes more sense for required but non-persistent files to be written there.~~

EDIT: I hadn't noticed that `~/.local/share/wezterm` isn't where the sock file is written on linux.

**Reply by khanra17** [reply]
setting $XDG_DATA_HOME doesn't work as well.
I wan't it anywhere except home dir

**Reply by tpoliaw** [reply]
Looking into the source, [it uses `dirs_next` for its version of `runtime_dir`, falling back to `$HOME/.local/share` if it's not found](https://github.com/wez/wezterm/blob/11dec45f08e3c7a611f59b030d4a9bd391807604/config/src/config.rs#L1665). From [the docs](https://docs.rs/dirs-next/2.0.0/dirs_next/fn.runtime_dir.html) it looks like it's never going to return anything for windows so the sock directory won't be configurable. 

It might be possible to mess around with the `$HOME` value it uses if it's possible to affect [`{FOLDERID_Profile}`](https://docs.rs/dirs-next/2.0.0/dirs_next/fn.home_dir.html) but I'd be surprised if that didn't cause all sorts of other problems.
